ckanext/unaids/tests/test_dataset_transfer.py

```diff
@@ -136,12 +139,11 @@ def test_send_dataset_transfer_emails_errors(self, app):
             type='test-schema',
             org_to_allow_transfer_to=org_2['id']
         )
-        expected_error = r'All DatasetTransferRequest emails failed *'
-        with pytest.raises(AssertionError, match=expected_error):
-            send_dataset_transfer_emails(
-                dataset_id=dataset['id'],
-                recipient_org_id=org_2['id']
-            )
+        emails_succeeded = send_dataset_transfer_emails(
+            dataset_id=dataset['id'],
+            recipient_org_id=org_2['id']
+        )
+        assert not emails_succeeded
```

Comment:
You're not checking that not e-mails got send out in this case. In order to do that you'll need two different mocks.
I suggest using it with `with` keyword instead of annotation:
```
with mock.patch('ckan.lib.mailer.mail_user') as mail_user_mock:
     [test code]
```
Also this could be split into two tests:
1. Which tests if e-mail with proper text are send on dataset transfer request
2. Verify if e-mail are not sent twice.